src/components/ui/Text/Text.tsx

```diff
@@ -8,16 +8,29 @@ import { customClassSwitcher } from '~/core';
 
 const COMPONENT_NAME = 'Text';
 
+const tagMap : { [key: string]: string } = {
+    div: 'div',
+    span: 'span',
+    p: 'p',
+    label: 'label',
+  };
+
 export type TextProps = {
     children: React.ReactNode;
     customRootClass?: string;
     className?: string;
+    as?: string;
 } & React.ComponentProps<'p'>;
 
-const Text = ({ children, customRootClass = '', className = '', ...props }: TextProps) => {
+const Text = ({ children, customRootClass = '', className = '', as='p' , ...props }: TextProps) => {
     const rootClass = customClassSwitcher(customRootClass, COMPONENT_NAME);
-
-    return <p className={`${rootClass} ${className}`} {...props}>{children}</p>;
+    
+      
+      const Tag = tagMap[as] || 'p';
+      
+      return React.createElement(Tag, { className: `${rootClass} ${className}`, ...props }, children) 
```

Comment:
You can use `as` directly. The default value for `as` should not be removed. Otherwise, you could put it like
```
return React.createElement(as ?? defaultTag, ...)
```

Reply:
![image](https://github.com/user-attachments/assets/5ae812fe-1ad8-4cb1-8a43-3edb5d214da5)
Using `as` directly doesn't work with faulty or required tags